src/tilda.c

```diff
@@ -436,68 +422,122 @@ static gboolean parse_cli (int argc, char *argv[], gchar **config_file)
         exit (EXIT_FAILURE);
     }
 
-    /* Now set the options in the config, if they changed */
-    if (background_color != config_getstr ("background_color")) {
-        config_setstr ("background_color", background_color);
+    /* TRUE if we should show the config wizard, FALSE otherwize */
+    return cli_opts->show_config;
+}
+
+/**
+ * Initialize a structure in which command-line parameters will be stored.
+ * @return a pointer to that structure
+ */
+struct tilda_cli_opts *init_cli_opts_struct()
+{
+    struct tilda_cli_opts *tco = malloc(sizeof(struct tilda_cli_opts));
```

Comment:
I do not see this being freed any where. I also do not like the name `tco`, I would prefer to name this `options`, or some other descriptive name. Also if you use `g_malloc0` from glib, then you get memory that has been zeroed and you can omit the whole block of setting members to NULL and 0 below.